### PR TITLE
Added support for ISO8601 timestamps with time zones

### PR DESCRIPTION
**Pull request by sjauld**
Ref #145 

http://jsonapi.org/recommendations/#date-and-time-fields
https://www.w3.org/TR/NOTE-datetime

### CODE REVIEW

File: constants.go [hunk 1]
```diff
@@ -11,7 +11,8 @@ const (
 	annotationISO8601   = "iso8601"
 	annotationSeperator = ","
 
-	iso8601TimeFormat = "2006-01-02T15:04:05Z"
+	iso8601TimeFormat         = "2006-01-02T15:04:05Z"
+	iso8601ExpandedTimeFormat = "2006-01-02T15:04:05-07:00"
```
Comment (hypnoglow) [inline constants.go]: I'm curious why not just use `time.RFC3339`?

Seems to work fine https://play.golang.org/p/1Rom5WsfCYO

Comment (sjauld) [reply constants.go]: Hmmm, I thought that I tried that and it didn't work. But yes that's a much better solution if it does work.

**Comment by aren55555** [comment]
I don't believe this works with `20180725T033557Z` which is apparently also ISO 8601 compliant.

I'm more inclined to think this library should not take a stance on the actual wire format, but rather support any spec via custom types implementing the `json.Marshaler` and `json.Unmarshaler` interfaces.

**Comment by sjauld** [comment]
So maybe the support for iso8601 should be removed altogether since it is incomplete and undocumented?

**Comment by aren55555** [comment]
@sjauld IMO that's the most preferred path forward. I'd also suggest an additional README to call out how to custom types to support IS08601 or even RFC3339

**Pull request closed without merging**